stm32f1: adc: Add functions to get and clear flags

cua-cua opened the pull request:
- Include new functions to get and clear flags from ADC status register (ADC_SR) on STM32F1 ADCs.
These are based on the timer functions:  timer_clear_flag and timer_get_flag for similar purposes. (Check stm32/common/timer_common_all.h and /stm32/common/timer_common_all.c)
- Add some descriptions and "@refgroup" on adc_common_v1.h file to get linked documentation.

karlp commented:
This is common code for all the -v1 parts.  Please add the functions to the -v1 code, not the f1 code. If you are going to expand the doxygen as well, please note the extended bits are defined for l1 and f4, along with the standard bits in the -v1 header file.

cua-cua commented:
Ok, so I added the code on adc_common_v1 files and deleted it from f1 ones.
I included extended bits on l1 and f4 headers and tried to do the comments similar to each style file.

cua-cua commented:
@karlp friendly ping 😇

karlp commented:
tahks for this, it's merged (finally) in 889b7de0
